src/utils/ucx/ucx_utils.h

```diff
@@ -126,6 +118,18 @@ class nixlUcxWorker {
 
     void reqRelease(nixlUcxReq req);
     void reqCancel(nixlUcxReq req);
+
+    /* Memory management */
+    static int memReg(std::shared_ptr<nixlUcxContext> &ctx, void *addr, size_t size, nixlUcxMem &mem);
+    static size_t packRkey(std::shared_ptr<nixlUcxContext> &ctx, nixlUcxMem &mem, uint64_t &addr, size_t &size);
+    static void memDereg(std::shared_ptr<nixlUcxContext> &ctx, nixlUcxMem &mem);
+
```

Comment:
shall we move these functions to nixlUcxContext?

Reply:
Basically that would require making nixlUcxContext to be friend class of nixlUcxMem and nixlUcxRkey due to the current design (all their data is private and is only accessible to their friend class nixlUcxWorker). I personally don't have a strong opinion on this but Artem suggested he would prefer to make them static and leave them in nixlUcxWorker as opposed to extending friend classes.

Reply:
Actually, I don't like the static functions approach much, and maybe friend is better. 

My reasoning:
* As mem, ep, etc. are handles, it's not right to expose them to the outside world.
* On the other hand, UCX components are tightly coupled - memory is registered with UCP context and later used to post with UCP endpoint which is derived from UCP worker.

So, in my opinion, the friend approach would be the most natural and easiest to implement.
Getters will intro unnecessary code without benefits.

Reply:
Moved rkey import functions to nixlUcxEp and memory registration functions to nixlUcxContext.